linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/GenIncOptimizer.scala

```diff
@@ -115,17 +102,22 @@ abstract class GenIncOptimizer private[optimizer] (config: CommonPhaseConfig) {
       }
 
       val newLinkedClasses = for (linkedClass <- unit.classDefs) yield {
-        def defs(container: Option[MethodContainer]) =
-          container.fold[List[Versioned[MethodDef]]](Nil)(_.optimizedDefs)
-
         val encodedName = linkedClass.encodedName
-        val memberNamespace =
-          if (linkedClass.kind == ClassKind.Interface) getDefaults(encodedName)
-          else getClass(encodedName)
+        val staticLikeContainers = staticLikes(encodedName)
+
+        val publicContainer = classes.get(encodedName).getOrElse {
+          staticLikeContainers(MemberNamespace.Public.ordinal)
```

Comment:
I think this warrants a comment: IIUC this happens for default methods?

Maybe even better than a comment would be an `assert(linkedClass.kind == Interface)`?

Reply:
I added a comment, and an assert. The assert is actually a bit more complex than what you suggest, as this code is called for all non-Scala-classes. It's just that the containers must be empty for the ones that are not interfaces.